android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirTaskExpireJob.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.task
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.google.android.fhir.FhirEngine
+import dagger.hilt.android.EntryPointAccessors
+import java.util.concurrent.TimeUnit
+import org.smartregister.fhircore.engine.di.FhirTaskExpireEntryPoint
+import org.smartregister.fhircore.engine.util.SharedPreferencesHelper
+
+class FhirTaskExpireJob constructor(val context: Context, workerParams: WorkerParameters) :
+  CoroutineWorker(context, workerParams) {
+
+  var fhirEngine: FhirEngine = getFhirEngine(context)
+  var fhirTaskExpireUtil: FhirTaskExpireUtil = getFhirTaskExpireUtil(context)
```

Comment:
It's possible to inject dependencies to `CoroutineWorker`. See [AppSyncWorker](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/sync/AppSyncWorker.kt) for example.

Reply:
I believe so but I spent a lot of time trying it but couldn't get it to work. 

You can possibly help to get this working. The current injection works almost similar and depends on Hilt